[ao][sparsity][fx] adding unit tests for sparsity+reference quant

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82204
* __->__ #81994

Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][sparsity][fx] adding unit tests for sparsity+reference quant

Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] adding unit tests for sparsity+reference quant"


Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/ao/sparsity/test_composability.py:
```diff
@@ -377,6 +377,59 @@ def test_q_prep_fx_before_s_prep(self):
         )
         self.assertGreaterAlmostEqual(cur_sparsity, sparse_config[0]["sparsity_level"])
 
+    def test_q_prep_fx_s_prep_ref_conv(self):
```
same for these, can we add some docstrings, it might get confusing at a later time

jerryzh168 commented on test/ao/sparsity/test_composability.py:
```diff
@@ -377,6 +377,59 @@ def test_q_prep_fx_before_s_prep(self):
+        mod = convert_to_reference_fx(mod)
+
+        # check that final module is the expected quantized module and that the model runs
+        self.assertTrue(isinstance(fqn_to_module(mod, "5"), torch.nn.intrinsic.LinearReLU))
```
I think we also want to check the 0th element of this module is a reference linear module as well, something like:
```
self.assertTrue(isinstance(fqn_to_module(mod, "5")[0], torch.nn.quantized._reference.Linear))
```

jerryzh168 commented on test/ao/sparsity/test_composability.py:
```diff
@@ -416,3 +469,42 @@ def test_s_prep_before_q_prep_fx(self):
+        mod(example)
+        mod = convert_to_reference_fx(mod)
+        # check that final module is the expected quantized module and that the model runs
+        self.assertTrue(isinstance(fqn_to_module(mod, "5"), torch.nn.intrinsic.LinearReLU))
```
same for this one

HDCharles committed:
Update on "[ao][sparsity][fx] adding unit tests for sparsity+reference quant"


Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] adding unit tests for sparsity+reference quant"


Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g

HDCharles committed:
Update on "[ao][sparsity][fx] adding unit tests for sparsity+reference quant"


Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] adding unit tests for sparsity+reference quant"


Summary: No functional changes here, the previous PRs handle the
necessary changes for this to work.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g
